task4/task4.ipynb

```diff
@@ -76,26 +99,19 @@
   },
   {
    "cell_type": "code",
-   "execution_count": 40,
+   "execution_count": 47,
    "metadata": {},
    "outputs": [
     {
      "name": "stdout",
      "output_type": "stream",
      "text": [
-      "[0.48308715 0.05753459 0.01415235 0.00602463 0.0026858  0.00161148\n",
-      " 0.00053716 0.00053716 0.00053716 0.00053716]\n",
-      "0.7060732139754536\n"
-     ]
-    },
-    {
-     "name": "stderr",
-     "output_type": "stream",
-     "text": [
-      "/opt/homebrew/lib/python3.10/site-packages/numpy/lib/shape_base.py:379: RuntimeWarning: Mean of empty slice\n",
-      "  res = asanyarray(func1d(inarr_view[ind0], *args, **kwargs))\n",
-      "/opt/homebrew/lib/python3.10/site-packages/numpy/lib/shape_base.py:402: RuntimeWarning: Mean of empty slice\n",
-      "  buff[ind] = asanyarray(func1d(inarr_view[ind], *args, **kwargs))\n"
+      "[       nan 0.48308715 0.11909775 0.24597981 0.42569815 0.44580342\n",
```

Comment:
Active probability of the first period is apparently defined as 1, even when the retention rate is NaN